I2C using deprecated Zephyr API

When you build samples involving I2C like samples/I2C.js, you will see warnings that it's using deprecated Zephyr structs:

/home/jimmy/projects/zephyrjs/zephyr.js/src/zjs_i2c_handler.c: In function ‘zjs_i2c_handle_open’:
/home/jimmy/projects/zephyrjs/zephyr.js/src/zjs_i2c_handler.c:32:19: warning: ‘dev_config’ is deprecated [-Wdeprecated-declarations]
             union dev_config cfg;
                   ^~~~~~~~~~
In file included from /home/jimmy/projects/zephyrjs/zephyr.js/src/zjs_i2c_handler.h:6:0,
                 from /home/jimmy/projects/zephyrjs/zephyr.js/src/zjs_i2c_handler.c:4:
/home/jimmy/projects/zephyrjs/zephyr.js/deps/zephyr/include/i2c.h:97:20: note: declared here
 union __deprecated dev_config {
                    ^~~~~~~~~~

--- a/src/zjs_i2c_handler.c
+++ b/src/zjs_i2c_handler.c
@@ -1,4 +1,4 @@
-// Copyright (c) 2016-2017, Intel Corporation.
+// Copyright (c) 2016-2018, Intel Corporation.
 
 // ZJS includes
 #include "zjs_i2c_handler.h"
@@ -27,14 +27,9 @@ int zjs_i2c_handle_open(u8_t msg_bus)
         if (!i2c_device[msg_bus]) {
             ERR_PRINT("I2C bus %s not found.\n", bus);
         } else {
-            /* TODO remove these hard coded numbers
-             * once the config API is made */
-            union dev_config cfg;
-            cfg.raw = 0;
-            cfg.bits.use_10_bit_addr = 0;
-            cfg.bits.speed = I2C_SPEED_STANDARD;
-            cfg.bits.is_master_device = 1;
-            error_code = i2c_configure(i2c_device[msg_bus], cfg.raw);
+            u32_t cfg;
+            cfg = I2C_SPEED_SET(I2C_SPEED_STANDARD) | I2C_MODE_MASTER;
+            error_code = i2c_configure(i2c_device[msg_bus], cfg);
 
             if (error_code != 0) {
                 ERR_PRINT("I2C bus %s configure failed with %i.\n", bus,

